### PR TITLE
NPE in FeedbackContributionQuestionDetails #4803

### LINKED ISSUE #4803: NPE in FeedbackContributionQuestionDetails
java.lang.NullPointerException
	at java.util.ArrayList.(ArrayList.java:164)
	at teammates.common.datatransfer.FeedbackContributionQuestionDetails.getTeamMembersEmail(FeedbackContributionQuestionDetails.java:568)
	at teammates.common.datatransfer.FeedbackContributionQuestionDetails.getTeamEvalResults(FeedbackContributionQuestionDetails.java:471)
	at teammates.common.datatransfer.FeedbackContributionResponseDetails.getContribQnTeamEvalResult(FeedbackContributionResponseDetails.java:207)
	at teammates.common.datatransfer.FeedbackContributionResponseDetails.getContributionQuestionResponseAnswerHtml(FeedbackContributionResponseDetails.java:81)
	at teammates.common.datatransfer.FeedbackContributionResponseDetails.getAnswerHtml(FeedbackContributionResponseDetails.java:65)
	at teammates.common.datatransfer.FeedbackSessionResultsBundle.getResponseAnswerHtml(FeedbackSessionResultsBundle.java:884)
	at teammates.ui.controller.InstructorFeedbackResultsPageData.buildResponseRowsForQuestion(InstructorFeedbackResultsPageData.java:1229)
	at teammates.ui.controller.InstructorFeedbackResultsPageData.buildQuestionTableAndResponseRows(InstructorFeedbackResultsPageData.java:1045)
	at teammates.ui.controller.InstructorFeedbackResultsPageData.buildQuestionTableAndResponseRows(InstructorFeedbackResultsPageData.java:1001)
	at teammates.ui.controller.InstructorFeedbackResultsPageData.initForViewByQuestion(InstructorFeedbackResultsPageData.java:153)
	at teammates.ui.controller.InstructorFeedbackResultsPageAction.execute(InstructorFeedbackResultsPageAction.java:130)
	at teammates.ui.controller.Action.executeAndPostProcess(Action.java:337)
	at teammates.ui.controller.ControllerServlet.doPost(ControllerServlet.java:59)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:637)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
	at teammates.storage.datastore.DatastoreFilter.doFilter(DatastoreFilter.java:28)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.appstats.AppstatsFilter.doFilter(AppstatsFilter.java:143)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.ParseBlobUploadFilter.doFilter(ParseBlobUploadFilter.java:125)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.runtime.jetty.SaveSessionFilter.doFilter(SaveSessionFilter.java:37)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.JdbcMySqlConnectionCleanupFilter.doFilter(JdbcMySqlConnectionCleanupFilter.java:60)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:43)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
	at com.google.apphosting.runtime.jetty.AppVersionHandlerMap.handle(AppVersionHandlerMap.java:260)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
	at com.google.apphosting.runtime.jetty.RpcRequestParser.parseAvailable(RpcRequestParser.java:78)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at com.google.apphosting.runtime.jetty.JettyServletEngineAdapter.serviceRequest(JettyServletEngineAdapter.java:148)
	at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.run(JavaRuntime.java:468)
	at com.google.tracing.TraceContext$TraceContextRunnable.runInContext(TraceContext.java:437)
	at com.google.tracing.TraceContext$TraceContextRunnable$1.run(TraceContext.java:444)
	at com.google.tracing.CurrentContext.runInContext(CurrentContext.java:256)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContextNoUnref(TraceContext.java:308)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContext(TraceContext.java:300)
	at com.google.tracing.TraceContext$TraceContextRunnable.run(TraceContext.java:441)
	at com.google.apphosting.runtime.ThreadGroupPool$PoolEntry.run(ThreadGroupPool.java:235)
	at java.lang.Thread.run(Thread.java:745)

### PR DESCRIPTION
**Pull request by kanghj**
Fixes #4803.

Running the test without the fix results in an error on the results page. This is a quick hotfix, @bamboo3250 should have a a deeper look soon.

@LowWeiLin, are you around to review this? or pick someone else.

### CODE REVIEW

File: src/test/resources/data/InstructorFeedbackResultsPageUiTest.json [hunk 1]
```diff
@@ -159,6 +173,24 @@
             "comments":"This student's name is Emily",
             "team":"Team 3",
             "section":"Section B"
+        },
+        "Sanitized.Alice":{
+            "googleId":"CFResultsUiT.alice.b",
+            "email":"CFResultsUiT.alice.b@gmail.tmt",
+            "course":"CFResultsUiT.SanitizedTeam",
+            "name":"Alice Betsy",
```
Comment (bamboo3250) [inline src/test/resources/data/InstructorFeedbackResultsPageUiTest.json]: How about we add `'` in the name as well?

File: src/test/resources/data/InstructorFeedbackResultsPageUiTest.json [hunk 2]
```diff
@@ -233,6 +265,24 @@
             "feedbackSessionType":"STANDARD",
             "sentOpenEmail":true,
             "sentPublishedEmail":true
+        },
+        "Session with sanitized data":{
+            "feedbackSessionName":"Sanitized Session",
```
Comment (bamboo3250) [inline src/test/resources/data/InstructorFeedbackResultsPageUiTest.json]: Likewise for session name.

**Comment by bamboo3250** [comment]
In StudentAttributes, there are only 4 fields `name`, `lastName`, `comments` and `team` that need special attention. I will plan to strengthen the test data for them now.

File: src/test/resources/data/InstructorFeedbackResultsPageUiTest.json (hunk 1 above)
Comment (kanghj) [inline src/test/resources/data/InstructorFeedbackResultsPageUiTest.json]: Ok, will add. It's not really related to this issue though

Comment (kanghj) [inline src/test/resources/data/InstructorFeedbackResultsPageUiTest.json]: @bamboo3250, added. Have a look again?

File: src/main/java/teammates/common/datatransfer/FeedbackSessionResultsBundle.java [hunk 3]
```diff
@@ -934,6 +934,15 @@ public String getTeamNameForEmail(String email) {
...
     }
+    
+    public String getUnsanitizedTeamNameForEmail(String email) {
+        String teamName = emailTeamNameTable.get(email);
+        if (teamName == null || email.equals(Const.GENERAL_QUESTION)) {
+            return Const.USER_NOBODY_TEXT;
+        } else {
+            return teamName;
+        }
+    }
```
Comment (damithc) [inline src/main/java/teammates/common/datatransfer/FeedbackSessionResultsBundle.java]: Why is sanitization being mention in a result bundles, question details, etc? These two things are totally unrelated. Need to clean up the sanitization in due course.

File: src/main/java/teammates/common/datatransfer/FeedbackContributionQuestionDetails.java [hunk 4]
```diff
@@ -576,8 +576,9 @@ public String getQuestionResultStatisticsCsv(
             FeedbackSessionResultsBundle bundle) {
         List<String> teamNames = new ArrayList<String>();
         for(FeedbackResponseAttributes response : responses){
-            if(!teamNames.contains(bundle.getTeamNameForEmail(response.giverEmail))){
-                teamNames.add(bundle.getTeamNameForEmail(response.giverEmail));
+            String teamNameOfResponseGiver = bundle.getUnsanitizedTeamNameForEmail(response.giverEmail);
+            if (!teamNames.contains(teamNameOfResponseGiver)) {
```
Comment (damithc) [inline src/main/java/teammates/common/datatransfer/FeedbackContributionQuestionDetails.java]: Introduce a variable to explain the meaning of this condition.

Comment (damithc) [inline src/main/java/teammates/common/datatransfer/FeedbackContributionQuestionDetails.java]: Ok, I understood the meaning after reading the rest of the code. May you should have used a `Set` for this? Not related to this issue, so no need to change.

File: src/main/java/teammates/common/datatransfer/FeedbackSessionResultsBundle.java (hunk 3 above)
Comment (kanghj) [inline src/main/java/teammates/common/datatransfer/FeedbackSessionResultsBundle.java]: Ah, yes. This is a temporary method that should be removed when a more systematic cleanup is done. The original method was used in other places, so I didn't want to affect behaviour anywhere else yet. I'll add a TODO here.

**Comment by kanghj** [comment]
Will merge to release as V5.67.01

**Comment by damithc** [comment]
We'll probably have to release this as V5.68 (because admin activity page doesn't work with patch versions)

**Comment by kanghj** [comment]
Got it. Merged to release as V5.68.

**Comment by thyageshm** [comment]
@damithc did you mean V5.68 or V5.67? the current milestone is 68.
